static/app/utils/replays/types.tsx

```diff
@@ -264,19 +264,19 @@ export function isPaintFrame(frame: SpanFrame): frame is PaintFrame {
 
 export function isDeadClick(frame: SlowClickFrame) {
   return (
-    ['a', 'button', 'input'].includes(frame.data.node?.tagName.toLowerCase() ?? '') &&
-    frame.data.endReason === 'timeout'
+    ['a', 'button', 'input'].includes(frame.data?.node?.tagName.toLowerCase() ?? '') &&
+    frame.data?.endReason === 'timeout'
   );
 }
 
 export function isDeadRageClick(frame: SlowClickFrame) {
   return Boolean(
-    isDeadClick(frame) && frame.data.clickCount && frame.data.clickCount >= 5
+    isDeadClick(frame) && frame.data?.clickCount && frame.data.clickCount >= 5
   );
 }
 
 export function isRageClick(frame: MultiClickFrame) {
-  return frame.data.clickCount >= 5;
+  return frame.data?.clickCount >= 5;
 }
```

Comment:
these need to have something like `: frame is DeadClickFrame` as part of the method signature to refine the types better.

is `data` really optional on the types? i expect to see some type changes to match all these added `?`

Reply:
kk

i do see the type fully defined in the sdk:
```
interface ReplayClickFrame extends ReplayBaseBreadcrumbFrame {
    category: 'ui.click';
    message: string;
    data: ReplayClickFrameData;
}
```

the stuff inside `data` is optional, but data should be there
i'm worried these question marks might get auto-removed if we add more type aware linting.

Reply:
I've added some code comments, hopefully they get read, but also not end of world if it does get removed since this is a small occurrence. (and ideally this gets fixed in SDK).